evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -70,13 +70,10 @@ def transform(self, X, y=None):
         Returns:
             pd.DataFrame: Transformed X
         """
-        X = _convert_to_woodwork_structure(X)
-        X = _convert_woodwork_types_wrapper(X.to_dataframe())
-        # Convert None to np.nan, since None cannot be properly handled
-        X = X.fillna(value=np.nan)
-
+        ww_X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(ww_X.to_dataframe())
         # Return early since bool dtype doesn't support nans and sklearn errors if all cols are bool
-        if (X.dtypes == bool).all():
+        if set(ww_X.logical_types.values()) == {ww.logical_types.Boolean} and X.isnull().sum().sum() == 0:
             return X
         X_null_dropped = X.copy()
         X_null_dropped.drop(self._all_null_cols, axis=1, errors='ignore', inplace=True)
```

Comment:
Hm. The following is specific to the `category`-typed pandas columns passed to the imputer, so I'm not sure there's much that Woodwork can do to help out, given we have to pass a pandas DF / numpy array to the component object.